machine.go

```diff
@@ -303,20 +270,26 @@ func (m *Machine) startVMM(ctx context.Context) (<-chan error, error) {
 			m.logger.Printf("Caught signal %s", sig)
 			m.cmd.Process.Signal(sig)
 		case err = <-vmchan:
-			exitCh <- err
+			m.errCh <- err
 		}
 	}()
 
 	// Wait for firecracker to initialize:
-	err = m.waitForSocket(3*time.Second, exitCh)
+	err = m.waitForSocket(3*time.Second, m.errCh)
 	if err != nil {
 		msg := fmt.Sprintf("Firecracker did not create API socket %s: %s", m.cfg.SocketPath, err)
 		err = errors.New(msg)
-		return exitCh, err
+		return err
 	}
 
 	m.logger.Debugf("returning from startVMM()")
-	return exitCh, nil
+	return nil
+}
+
+// Wait will wait for the microVM to finish execution or if an error was
+// encountered.
+func (m *Machine) Wait() error {
```

Comment:
Would it be worth having `Wait` take a `context.Context` so it can be canceled?

Reply:
After some thought, I think it makes a lot of sense to do this. Originally I was wondering if we should follow the same signature as the `exec.Cmd` wait call, but due to the difference in complexity I think it makes sense to pass a context in.

Reply:
I think so too.  It would be possible to wrap the call to `Wait` like the following:

```go
ctx, _ := context.WithTimeout(context.Background(), timeout)
errCh := make(chan errCh)
go func(){
	errCh <- m.Wait()
}()
select {
	case <- ctx.Done():
		return errors.New("timed out")
	case err <- errCh:
		return err
}
```

But this is a lot of code, and it leaves a dangling goroutine.  The risk of the dangling goroutine is that if you then call `Wait` again, it's nondeterministic which one will actually get the error and return, and which one will block.